pyvrp/cpp/search/SwapStar.cpp

```diff
@@ -183,32 +207,28 @@ Cost SwapStar::evaluate(Route *routeU,
 {
     best = {};
 
-    if (updated[routeV->idx()])
-    {
-        updateRemovalCosts(routeV, costEvaluator);
-        updated[routeV->idx()] = false;
-
-        for (size_t idx = data.numDepots(); idx != data.numLocations(); ++idx)
-            cache(routeV->idx(), idx).shouldUpdate = true;
-    }
-
-    if (updated[routeU->idx()])
-    {
-        updateRemovalCosts(routeU, costEvaluator);
-        updated[routeV->idx()] = false;
+    for (auto *route : {routeU, routeV})
+        if (updated[route->idx()])
+        {
+            updateRemovalCosts(route, costEvaluator);
+            updated[route->idx()] = false;
 
-        for (size_t idx = data.numDepots(); idx != data.numLocations(); ++idx)
-            cache(routeU->idx(), idx).shouldUpdate = true;
-    }
+            for (size_t idx = 0; idx != data.numLocations(); ++idx)
+                cache(route->idx(), idx).shouldUpdate = true;
+        }
 
     for (auto *U : *routeU)
         for (auto *V : *routeV)
         {
             Cost deltaCost = 0;
 
-            auto const uDemand = data.location(U->client()).demand;
-            auto const vDemand = data.location(V->client()).demand;
-            auto const loadDiff = uDemand - vDemand;
+            // Approximates the actual load changes. It's pretty rough, but
+            // fast and seems to work really well.
+            auto const &uClient = data.location(U->client());
+            auto const &vClient = data.location(V->client());
+            auto const uLoad = std::max(uClient.demand, uClient.supply);
+            auto const vLoad = std::max(vClient.demand, vClient.supply);
+            auto const loadDiff = uLoad - vLoad;
 
             deltaCost += costEvaluator.loadPenalty(routeU->load() - loadDiff,
                                                    routeU->capacity());
```

Comment:
This is wrong, or rather, approximate. But if I do a full evaluation here that has a really strong impact on performance. So we may just want to keep this approximate, and then once a good move has been identified, leave it to `evaluateMove` to determine the exact costs (`evaluateMove` has been updated to compute the correct costs).